src/Servers/Kestrel/Transport.Sockets/src/Internal/IOQueue.cs

```diff
@@ -73,4 +75,19 @@ public Work(Action<object?> callback, object? state)
             State = state;
         }
     }
+
+    private static int DetermineDefaultCount()
+    {
+        // Since each IOQueue schedules one work item to process its work, the number of IOQueues determines the maximum
+        // parallelism of processing work queued to IOQueues. The default number below is based on the processor count and tries
+        // to use a high-enough number for that to not be a significant limiting factor for throughput.
+
+        int processorCount = Environment.ProcessorCount;
+        if (OperatingSystem.IsWindows() || processorCount <= 32)
+        {
+            return Math.Min(processorCount, 16);
+        }
+
+        return (processorCount + 1) / 2;
```

Comment:
Why this and not just `processorCount / 2` given this branch will only be taken if `processorCount > 32`?